code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,352 @@
+				addtimer(CALLBACK(src, .proc/return_elite), 30)
+
+				
+obj/structure/elite_tumor/proc/spawn_elite(var/mob/dead/observer/elitemind)
+	var/selectedspawn = pick(potentialspawns)
+	mychild = new selectedspawn(loc)
+	visible_message("<span class='boldwarning'>[mychild] emerges from the tumor!</span>")
+	playsound(loc,'sound/effects/phasein.ogg', 200, 0, 50, TRUE, TRUE)
+	if(boosted)
+		mychild.key = elitemind.key
+		mychild.sentience_act()
+	INVOKE_ASYNC(src, .proc/arena_trap)
+
+obj/structure/elite_tumor/proc/return_elite()
+	mychild.forceMove(loc)
+	visible_message("<span class='boldwarning'>[mychild] emerges from the tumor!</span>")
+	playsound(loc,'sound/effects/phasein.ogg', 200, 0, 50, TRUE, TRUE)
+	mychild.revive(full_heal = TRUE, admin_revive = TRUE)
+	if(boosted)
+		mychild.maxHealth = mychild.maxHealth * 2
+		mychild.health = mychild.maxHealth
+		
+/obj/structure/elite_tumor/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/gps, "Menacing Signal")
+	START_PROCESSING(SSobj, src)
+	
+/obj/structure/elite_tumor/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	mychild = null
+	activator = null
+	return ..()
+	
+/obj/structure/elite_tumor/process()
+	if(isturf(loc))
+		for(var/mob/living/simple_animal/hostile/asteroid/elite/elitehere in loc)
+			if(elitehere == mychild && activity == TUMOR_PASSIVE)
+				mychild.adjustHealth(-mychild.maxHealth*0.05)
+				var/obj/effect/temp_visual/heal/H = new /obj/effect/temp_visual/heal(get_turf(mychild))
+				H.color = "#FF0000"
+		
+/obj/structure/elite_tumor/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(istype(I, /obj/item/organ/regenerative_core) && activity == TUMOR_INACTIVE && !boosted)
+		var/obj/item/organ/regenerative_core/core = I
+		if(core.preserved)
+			visible_message("<span class='boldwarning'>As [user] drops the core into the tumor, the tumor appears to swell.</span>")
+			icon_state = "advanced_tumor"
+			boosted = TRUE
+			light_range = 6
+			desc = "[desc]  This one seems to glow with a strong intensity."
+			qdel(core)
+			return TRUE
+			
+/obj/structure/elite_tumor/proc/arena_checks()
+	INVOKE_ASYNC(src, .proc/arena_trap)  //Gets another arena trap queued up for when this one runs out.
+	INVOKE_ASYNC(src, .proc/border_check)  //Checks to see if our fighters got out of the arena somehow.
+	INVOKE_ASYNC(src, .proc/fighters_check)  //Checks to see if our fighters died.
+			
+/obj/structure/elite_tumor/proc/arena_trap()
+	var/turf/T = get_turf(src)
+	if(T && activity == TUMOR_ACTIVE)
+		for(var/t in RANGE_TURFS(12, T))
+			if(t && get_dist(t, T) == 12)
+				var/obj/effect/temp_visual/elite_tumor_wall/newwall
+				newwall = new /obj/effect/temp_visual/elite_tumor_wall(t, src)
+				newwall.activator = src.activator
+				newwall.ourelite = src.mychild
+		addtimer(CALLBACK(src, .proc/arena_checks), 50)
+		
+/obj/structure/elite_tumor/proc/border_check()
+	if(activity == TUMOR_ACTIVE)
+		if(activator != null && !QDELETED(activator) && get_dist(src, activator) >= 12)
```

Comment:
You get the idea, you nest far too deep do a check on the entire pr yourself you nest deep all over it.

Reply:
OK